samcli/commands/init/__init__.py

```diff
@@ -286,9 +286,9 @@ def do_cli(
     image_bool = name and pt_explicit and base_image
     if location or zip_bool or image_bool:
         # need to turn app_template into a location before we generate
-        templates = InitTemplates(no_interactive)
+        templates = InitTemplates()
         if package_type == IMAGE and image_bool:
-            base_image, runtime = _get_runtime_from_image(base_image)
+            runtime = _get_runtime_from_image(base_image)
```

Comment:
`base_image` is used below, why we are not assigning a value for it?

Reply:
First the _get_runtime_from_image method is confusing as it does 2 things but named something else, inspired good coding practice it should return only the runtime derived from the the image name. This was why I refactored it to return just the runtime derived from the image name and moved other functionalities to the method _get_choice_from_options which would return image options for the customer to select. The option selected (image) is then passed to _get_runtime_from_image which gets the runtime. 

The `base_image` is passed to this method. it seemed redundant passing an argument to method only to return the same argument with manipulating it.